include/h2o/socket.h

```diff
@@ -108,6 +108,11 @@ struct st_h2o_socket_t {
      * total bytes written (above the TLS layer)
      */
     size_t bytes_written;
+    /**
+     * ternary flag to track if sock is being traced. 0 indicates value unknown, 1 means is traced, -1 is not traced
+     */
+    int _is_traced;
```

Comment:
I am not sure if this should be a ternary flag.

If I understand correctly, the intent of defining this as a ternary flag is to lazy-load the information from the eBPF map. However, I am not sure if that should be done, because the map is an LRU.

I think we should try to load the information as soon as the server-side of the socket is created, to avoid the risk of the entry corresponding to the created socket evicted from the eBPF map. The positive side effect of making such a change would be that this can then be an ordinary boolean flag, because the "unknown" state becomes unnecessary.

WDYT?

Reply:
> If I understand correctly, the intent of defining this as a ternary flag is to lazy-load the information from the eBPF map

Yes exactly - but the LRU relatively small data retention is a good point indeed !

My initial idea behind the lazily-loaded ternary state was to completely isolate the tracing path from the rest of the code.  `h2o_socket_is_traced` is then the only gateway to the tracing code, which is only executed once a probe is attached. 

This works if `h2o_socket_is_traced` isn't called too late after the arrival of the connection, so the relevant information is hopefully still within the map at this time.

I'm not sure what to conclude from here. I implemented the non-ternary-state version in this [commit](https://github.com/pldubouilh/h2o/commit/5dcee546cb4dd4ecc773cd7f32799ad56902ecc2) - feel free to tell me what you think, I can cherry-pick it to this branch if you think it's a better setup :+1:

Reply:
Thank you for the non-ternary-state version. Applied.